src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/X500RelativeDistinguishedName.cs

```diff
@@ -0,0 +1,153 @@
+            AsnValueReader rdn = outer.ReadSetOf(skipSortOrderValidation: true);
+            AsnValueReader typeAndValue = rdn.ReadSequence();
+
+            Oid firstType = Oids.GetSharedOrNewOid(ref typeAndValue);
+            ReadOnlySpan<byte> firstValue = typeAndValue.ReadEncodedValue();
+            typeAndValue.ThrowIfNotEmpty();
+
+            if (rdn.HasData)
+            {
+                while (rdn.HasData)
+                {
+                    typeAndValue = rdn.ReadSequence();
+
+                    if (Oids.GetSharedOrNullOid(ref typeAndValue) is null)
+                    {
+                        typeAndValue.ReadObjectIdentifier();
+                    }
+
+                    typeAndValue.ReadEncodedValue();
+                    typeAndValue.ThrowIfNotEmpty();
+                }
+            }
+            else
+            {
+                _singleElementType = firstType;
+
+                if (!rawDataSpan.Overlaps(firstValue, out int offset))
+                {
+                    Debug.Fail("AsnValueReader.ReadEncodedValue returned data outside its bounds");
+                    throw new UnreachableException();
+                }
+
+                _singleElementValue = rawData.Slice(offset, firstValue.Length);
+            }
+        }
+
+        /// <summary>
+        ///   Gets a value that indicates whether this Relative Distinguished Name is composed
+        ///   of multiple attributes or only a single attribute.
+        /// </summary>
+        /// <value>
+        ///   <see langword="true"/> if the Relative Distinguished Name is composed of multiple
+        ///   attributes; <see langword="false"/> if it is composed of only a single attribute.
+        /// </value>
+        public bool HasMultipleElements => _singleElementType is null;
+
+        /// <summary>
+        ///   Gets the object identifier (OID) identifying the single attribute value for this
+        ///   Relative Distinguished Name (RDN), when the RDN only contains one attribute.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">
+        ///   The Relative Distinguished Name has multiple attributes (<see cref="HasMultipleElements"/>
+        ///   is <see langword="true" />).
+        /// </exception>
+        public Oid GetSingleElementType()
+        {
+            if (_singleElementType is null)
+            {
+                throw new InvalidOperationException(SR.Cryptography_X500_MultiValued);
+            }
+
+            return _singleElementType;
+        }
+
+        /// <summary>
+        ///   Gets the textual representation of the value for the Relative Distinguished Name (RDN),
+        ///   when the RDN only contains one attribute.
+        /// </summary>
+        /// <returns>
+        ///   <see langword="null"/> if the encoded value is not a textual data type, such as an
+        ///   <c>OCTET STRING</c>; otherwise, the textual representation of the value.
```

Comment:
"is not a textual data type, such as an \<c>OCTET STRING\</c>"

To me this reads as OCTET STRING is an example of a textual data type.

Reply:
:shipit: